Sources/Testing/Running/Configuration.swift

```diff
@@ -203,3 +203,61 @@ public struct Configuration: Sendable {
   /// The test case filter to which test cases should be filtered when run.
   public var testCaseFilter: TestCaseFilter = { _, _ in true }
 }
+
+extension Configuration.RepetitionPolicy {
+
+  /// A serializable type describing whether or not, and how, to iterate a test
+  /// plan repeatedly.
+  ///
+  @_spi(ForToolsIntegrationOnly)
+  public struct Snapshot: Sendable, Codable {
+
+    /// The conditions under which test iterations should continue.
+    ///
+    /// If the value of this property is `nil`, a test plan will be run
+    /// ``count`` times regardless of whether or not issues are encountered
+    /// while running.
+    public var continuationCondition: ContinuationCondition.Snapshot?
+
+    /// The maximum number of times the test run should iterate.
+    ///
+    /// - Precondition: The value of this property must be greater than or equal
+    ///   to `1`.
+    public var maximumIterationCount: Int
+
+    public init(snapshotting repetitionPolicy: Configuration.RepetitionPolicy) {
+      continuationCondition = repetitionPolicy.continuationCondition
+        .map(Configuration.RepetitionPolicy.ContinuationCondition.Snapshot.init(snapshotting:))
+      maximumIterationCount = repetitionPolicy.maximumIterationCount
+    }
+  }
+}
+
+extension Configuration.RepetitionPolicy.ContinuationCondition {
+
+  /// An enumeration describing the conditions under which test iterations
+  /// should continue.
+  @_spi(ForToolsIntegrationOnly)
+  public enum Snapshot: Sendable, Codable {
```

Comment:
`ContinuationCondition` does not need a snapshot type. It can just be `Codable` directly as it is pure data.